<?xml version="1.0" ?>
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot name="x02">
    <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
    <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
    <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
    <group name="LeftLeg">
        <link name="l_sole" />
        <joint name="LHipYaw" />
        <joint name="LHipRoll" />
        <joint name="LHipPitch" />
        <joint name="LKnee" />
        <joint name="LAnklePitch" />
    </group>
    <group name="RightLeg">
        <link name="r_sole" />
        <joint name="RHipYaw" />
        <joint name="RHipRoll" />
        <joint name="RHipPitch" />
        <joint name="RKnee" />
        <joint name="RAnklePitch" />
    </group>
    <group name="Legs">
        <joint name="LHipYaw" />
        <joint name="LHipRoll" />
        <joint name="LHipPitch" />
        <joint name="LKnee" />
        <joint name="LAnklePitch" />
        <joint name="RHipYaw" />
        <joint name="RHipRoll" />
        <joint name="RHipPitch" />
        <joint name="RKnee" />
        <joint name="RAnklePitch" />
    </group>
    <group name="RightArm">
        <link name="r_wrist" />
        <joint name="RShoulderPitch" />
        <joint name="RShoulderRoll" />
        <joint name="RElbow" />
    </group>
    <group name="LeftArm">
        <link name="l_wrist" />
        <joint name="LShoulderPitch" />
        <joint name="LShoulderRoll" />
        <joint name="LElbow" />
    </group>
    <group name="Arms">
        <link name="r_wrist" />
        <link name="l_wrist" />
        <joint name="RShoulderPitch" />
        <joint name="RShoulderRoll" />
        <joint name="RElbow" />
        <joint name="LShoulderPitch" />
        <joint name="LShoulderRoll" />
        <joint name="LElbow" />
    </group>
    <group name="Head">
        <joint name="HeadPan" />
        <joint name="HeadTilt" />
    </group>
    <group name="All">
        <joint name="HeadPan" />
        <joint name="HeadTilt" />
        <joint name="LHipYaw" />
        <joint name="LHipRoll" />
        <joint name="LHipPitch" />
        <joint name="LKnee" />
        <joint name="LAnklePitch" />
        <joint name="LShoulderPitch" />
        <joint name="LShoulderRoll" />
        <joint name="LElbow" />
        <joint name="RHipYaw" />
        <joint name="RHipRoll" />
        <joint name="RHipPitch" />
        <joint name="RKnee" />
        <joint name="RAnklePitch" />
        <joint name="RAnkleRoll" />
        <joint name="RShoulderPitch" />
        <joint name="RShoulderRoll" />
        <joint name="RElbow" />
    </group>

    <group_state name="initial" group="LeftLeg">
        <joint name="LHipYaw" value="0.0" />
        <joint name="LHipRoll" value="0.0" />
        <joint name="LHipPitch" value="0.7" />
        <joint name="LKnee" value="1.0" />
        <joint name="LAnklePitch" value="-0.4" />
    </group_state>
    <group_state name="initial" group="RightLeg">
        <joint name="RHipYaw" value="0.0" />
        <joint name="RHipRoll" value="0.0" />
        <joint name="RHipPitch" value="-0.7" />
        <joint name="RKnee" value="-1.0" />
        <joint name="RAnklePitch" value="0.4" />
        <joint name="RAnkleRoll" value="0.0" />
    </group_state>
</robot>
